src/smolagents/models.py

```diff
@@ -367,6 +375,98 @@ def __call__(
         return message
 
 
+class VLLMModel(Model):
+    """This engine initializes a model and tokenizer from the given `model_id`.
+
+    Parameters:
+        model_id (`str`, *optional*, defaults to `"HuggingFaceTB/SmolLM2-1.7B-Instruct"`):
+            The Hugging Face model ID to be used for inference. This can be a path or model identifier from the Hugging Face model hub.
+    """
+
+    def __init__(self, model_id: Optional[str] = None):
+        super().__init__()
+
+        default_model_id = "HuggingFaceTB/SmolLM2-1.7B-Instruct"
+        if model_id is None:
+            model_id = default_model_id
+            logger.warning(
+                f"`model_id`not provided, using this default tokenizer for token counts: '{model_id}'"
+            )
+
+        self.model_id = model_id
+        self.model = LLM(model=model_id)
+        self.tokenizer = get_tokenizer(model_id)
+
+    def cleanup(self):
+        destroy_model_parallel()
+        if self.model is not None:
+            del self.model.llm_engine.model_executor.driver_worker
+        self.model = None
+        gc.collect()
+        destroy_distributed_environment()
+        torch.cuda.empty_cache()
+
+    def __call__(
+        self,
+        messages: List[Dict[str, str]],
+        stop_sequences: Optional[List[str]] = None,
+        grammar: Optional[str] = None,
+        max_tokens: int = 1500,
+        tools_to_call_from: Optional[List[Tool]] = None,
+    ) -> ChatMessage:
+        messages = get_clean_message_list(
+            messages, role_conversions=tool_role_conversions
+        )
+        if tools_to_call_from is not None:
+            prompt = self.tokenizer.apply_chat_template(
+                messages,
+                tools=[get_tool_json_schema(tool) for tool in tools_to_call_from],
+                add_generation_prompt=True,
+                tokenize=False,
+            )
+        else:
+            prompt = self.tokenizer.apply_chat_template(
+                messages,
+                tokenize=False,
+            )
+
+        sampling_params = SamplingParams(
+            n=1, temperature=0.3, max_tokens=max_tokens, stop=stop_sequences
+        )
```

Comment:
We might want to set this through args no? Typically something we could want to capture from the kwargs